File: Level #2/Advanced Javascript/03. PrototypeChainAndInheritance/02. 2DGeometryStructure-Prototypal/models/Rectangle.js
var shapes = shapes || {};

(function (scope) {
    function validateSides(width, height) {
        if (isNaN(width) || isNaN(height) || width <= 0 || height <= 0) {
            throw new Error('Rectangle sides should be numbers greater than 0.');
        }
    }

    var rectangle = scope.shape.extends({
        init: function init(leftCorner, width, height, color) {
            this._super.init.call(this, leftCorner, color);
            validateSides(width, height);
            this._width = width;
            this._height = height;
            this._POINT_TYPE = 'Top left corner';
            this._type = 'Rectangle';
            return this;
        },

        toString: function toString() {
            var rectangle = this._super.toString.call(this) +
                            'Width: ' + this._width + '\n' +
                            'Height: ' + this._height + '\n';

            return rectangle;
        }
    });

    scope.rectangle = rectangle;
}(shapes))

File: Level #2/OOP/Homework-Mates/8.Encapsulation and Polyphormism/1/EncapsulationAndPolymorphism/Shapes/Rectangle.cs
using System;

namespace Shapes
{
    class Rectangle : BasicShape
    {
        public Rectangle(double width, double height)
            : base(width, height)
        {
        }

        public override double CalculateArea()
        {
            return this.Width * this.Height;
        }

        public override double CalculatePerimeter()
        {
            return (this.Height + this.Width) * 2;
        }
    }
}
